dace/transformation/estimator/soap/solver.py

```diff
@@ -0,0 +1,100 @@
+from os import error, path
+import socket
+import time
+from subprocess import call
+from dataclasses import dataclass, field
+from typing import Dict
+import json
+from dace import Config
+# MATLAB SERVER CONFIGURATION 
+
+@dataclass
+class Solver():     
+    cached_only : bool = False
+    caching_solutions: bool = True
+    debug: bool = False
+    conn : socket = field(default_factory = socket.socket, hash = False)
+    status: str = "disconnected"
+    solver_cache: Dict = field(default_factory=dict)
+
+    def __post_init__(self):
+        self.cached_only = Config.get("soap", "solver", "only_db") 
+        self.caching_solutions = Config.get("soap", "solver", "caching_solver_solutions") 
+    
+    def start_solver(self):
+        if self.caching_solutions:
+            if path.getsize(Config.get("soap", "solver", "db_path")) > 0:
+                with open(Config.get("soap", "solver", "db_path"), "r") as solver_cache_file:
+                    self.solver_cache = json.load(solver_cache_file)
+        if self.cached_only:
+            return
+        # configuration
+        port = 30000
+        if Config.get("soap", "solver", "remote_matlab") :
+            address = Config.get("soap", "solver", "remote_solver_address")
+        else:
+            address = 'localhost'
+            # start matlab in background
+            call("matlab.exe -nosplash -nodesktop -r \"cd('" + Config.get("soap", "solver", "local_solver_path") + 

```

Comment:
We need to add Linux and Mac compatible calls here. I have one that works for Linux but it kinda breaks the shell (bash) afterward:
```python
            try:
                call("matlab -nosplash -nodesktop -r \"cd('" + Config.get("soap", "solver", "local_solver_path")"'); BackgroundSolver(" + str(port) + ");exit\"", shell=True, timeout=1)
            except TimeoutExpired:
                pass
```